napari/_vispy/vispy_transforms.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from vispy.visuals.transforms import STTransform, ChainTransform
+
+
+class VispyTransformChain:
+    """Class containing an ordered sequence of vispy transforms.
+
+    Parameters
+    ----------
+    napari_transform_chain : napari.layers.transforms.TransformChain
+        Transform chain for an ordered sequence of napari transforms.
+
+    Attributes
+    ----------
+    scale : ndarray, shape(4)
+        Combined scale of all chained transforms, vispy format (X,Y,Z,T).
+    simplified_transform : vispy.visuals.transforms.ChainTransform
+        Simplified ChainTransform where transforms have been combined into one.
+        Used to reduce compute time, since a single matrix operation is faster.
+    transform_chain : vispy.visuals.transforms.ChainTransform
+        Transform chain for an ordered sequence of vispy transforms.
+    translate : ndarray, shape(4)
+        Combined tranlation of all chained transforms, vispy format (X,Y,Z,T).
+    """
+
+    def __init__(self, napari_transform_chain):
+        self.napari_transform_chain = napari_transform_chain
+
+        self.transform_chain = self._vispy_transform_chain()
+        self._calculate_attributes()
+
+        self.napari_transform_chain.events.added.connect(self._add)
+        self.napari_transform_chain.events.removed.connect(self._remove)
+        self.transform_chain.events.reordered.connect(self._reorder)
+
+    def _vispy_transform_chain(self):
+        """Builds vispy ChainTransform from a napari TransformChain."""
+        vispy_transforms = []
+        for t in self.napari_transform_chain:
+            # From napari format (t, z, y, x)
+            # to the vispy format (x, y, z, ?)
+            translate = np.flip(t.translate)  # vispy ordered array
+            scale = np.flip(t.scale)  # vispy ordered array
+            vispy_transforms.append(
+                STTransform(scale=scale, translate=translate)
+            )
+        self.transform_chain = ChainTransform(vispy_transforms)
+        return self.transform_chain
+
+    def _calculate_attributes(self):
+        """Simplified transformation matrix, scale, and transle attributes.
```

Comment:
All this attribute stuff is so I can try switching out the STTransform in `_master_transform` for an instance of our new VispyChainTransform class. I want to try running the test suite before making more changes (but I expect we will tidy this up later on).